torch/_inductor/lowering.py

```diff
@@ -1163,6 +1164,7 @@ def philox_rand_like(x, seed, offset):
     device = x.get_device()
     dtype = x.get_dtype()
     size = x.get_size()
+    offset = offset.data if type(offset) is GenericBox else offset
```

Comment:
With dynamic shapes, this will be wrapped into `GenericBox`, otherwise not. Different code paths are used.